core/src/avm2/activation.rs

```diff
@@ -2898,34 +2898,62 @@ impl<'a, 'gc> Activation<'a, 'gc> {
     }
 
     fn op_as_type_late(&mut self) -> Result<FrameControl<'gc>, Error<'gc>> {
-        let class = self
-            .pop_stack()
-            .as_object()
-            .and_then(|c| c.as_class_object())
-            .ok_or("Cannot coerce a value to a type that is null, undefined, or not a class")?;
-        let value = self.pop_stack();
+        let class = self.pop_stack();
 
-        if value.is_of_type(self, class) {
-            self.push_stack(value);
-        } else {
-            self.push_stack(Value::Null);
+        if matches!(class, Value::Undefined) {
```

Comment:
can we check for `Value::Null | Value::Undefined` in one go here?

Reply:
`undefined` throws the undefined error, while `null` and all other primitives throw the null error.

Reply:
Hmm, ok, I see it; kinda weird ;D
Anyway, if I wrote it, I'd still check for `Null|Undefined` as a pattern, but if you don't wanna, that's okay too.